adapter/gin/middleware.go

```diff
@@ -0,0 +1,39 @@
+package gin
+
+import (
+	"net/http"
+
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/gin-gonic/gin"
+)
+
+// SentinelMiddleware
+func SentinelMiddleware(opts ...Option) gin.HandlerFunc {
+	options := evaluateOptions(opts)
+	return func(c *gin.Context) {
+		resourceName := c.Request.Method + ":" + c.Request.URL.Path
```

Comment:
Maybe here we need `c.FullPath()` to get the matched route path (e.g. `/foo/:id`) rather than the actual path (to avoid generating too many URL entries, bringing memory footprint). And as the comments say:

> FullPath returns a matched route full path. For not found routes returns an empty string.

So when the `FullPath()` returns the empty string, we could just skip it.

Reply:
Actually we did this by customizing an extractor and matcher.
But in the latest version, the FullPath method is provided, so we can use it directly.

Thank you for pointing it out :)